### PR TITLE
F/farcaster r1

### PR DESCRIPTION
**Pull request by jordipainan**
### Summary by CodeRabbit

- New Feature: Added support for Farcaster tokens in the census creation process, including compatibility checks and transformations.
- New Feature: Introduced a new flag `farcaster` in the configuration struct to enable Farcaster functionality.
- New Feature: Added a new token type 'farcaster' to the database and updated SQL queries to handle this new type.
- New Feature: Created a new `farcaster` package with database operations and user management functions.
- New Feature: Added a `CensusKeys` method to various provider structs to transform data based on the provider's strategy.
- New Feature: Included support for Farcaster contract types and log topics in the Web3Provider.

### CODE REVIEW

**Comment by jordipainan** [comment]
squash commits please

File: api/censuses.go [hunk 1]
```diff
@@ -231,6 +231,52 @@ func (capi *census3API) createAndPublishCensus(req *CreateCensusRequest, qID str
...
+	// 	if !ok {
+	// 		return nil, 0, fmt.Errorf("farcaster provider not configured")
+	// 	}
+	// 	farcasterHolders, err := provider.CensusKeys(holders)
+	// 	if err != nil {
+	// 		return nil, 0, err
+	// 	}
+	// 	return farcasterHolders, 100, nil
+	// }
+	// by default, return the same holders and an accuracy of 100%
```
Comment (lucasmenendez) [inline api/censuses.go:276]: this code is only required for token gated use case, right?

File: scanner/providers/farcaster/provider.go [hunk 2]
```diff
@@ -0,0 +1,549 @@
...
+		"progress", fmt.Sprintf("%d%%", (fromBlock*100)/toBlock),
+	)
+	p.contracts.keyRegistrySynced.Store(synced)
+	log.Debugf("found %d keys added for users registered in the key registry contract", len(addedKeys))
+	log.Debugf("found %d keys removed for users registered in the key registry contract", len(removedKeys))
+	return addedKeys, removedKeys, lastBlock, synced, nil
+}
+
+// Close method is not implemented for Farcaster Key Registry.
+func (p *FarcasterProvider) Close() error {
```
Comment (lucasmenendez) [inline scanner/providers/farcaster/provider.go:322]: TODO: Close the internal clients and databases

File: api/censuses.go (hunk 1 above)
Comment (jordipainan) [reply api/censuses.go:276]: yes right

**Pull request merged**